addons/escoria-core/game/core-scripts/esc/esc_event_manager.gd

```diff
@@ -372,16 +372,18 @@ func set_changing_scene(p_is_changing_scene: bool) -> void:
 # The event finished running
 #
 # #### Parameters
-# - finished_statement: statement object that finished
+# - finished_event: statement object representing the event that finished
+# - finished_statement: statement object representing the "deepest" statement (most likely a command)
+#   that just completed; this is useful for interrupted or failed statements especially
 # - return_code: Return code of the finished event
 # - channel_name: Name of the channel that the event came from
-func _on_event_finished(finished_statement: ESCStatement, return_code: int, channel_name: String) -> void:
+func _on_event_finished(finished_event: ESCStatement, finished_statement: ESCStatement, return_code: int, channel_name: String) -> void:
```

Comment:
Why isn't `finished_event` of type `ESCEvent` ?

Reply:
Because that's how `ESCStatement` works: When it's finished its `run()` method, it signals into the event manager using `self` as the `finished_event` argument. I don't think `ESCStatement` has a reference to an event or even shares the same structure.

This is partly what I'm talking about when I say "statement", "event", and "command" are confusing.

Another example: `ESCCommand` and `ESCBaseCommand` serve entirely different purposes.